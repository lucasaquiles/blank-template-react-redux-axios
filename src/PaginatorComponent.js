import React, { Component } from 'react';

export default class PaginatorComponent extends Component {
    
    handlerPaginatorClick = (event) => {
        event.preventDefault();
        console.log("event", event.currentTarget.getAttribute('current'));
        
        
    }

    componentDidMount(){
        this.props.pagitationAction(2);        
    }

    createPageList = () => {
        const url = "#";
        let list = [];
        
        for (let i = 1; i <= this.props.total; i++){
            
            list.push(<li key={i}><a href={url} onClick={this.handlerPaginatorClick} current={i}>{i}</a></li>);
        }
        return list;
    }

    render() {
        return (
            <ul class="paginador">
                 {this.createPageList()}
            </ul>
        )
    }
}