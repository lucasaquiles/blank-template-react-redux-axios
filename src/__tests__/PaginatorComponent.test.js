import React from 'react';
import { shallow, mount, render } from  'enzyme';
import configureStore from 'redux-mock-store';
import PaginatorComponent from '../PaginatorComponent';

const mockStore = configureStore();
const store = mockStore({});

describe('Paginator', () => {

    it('should render ', ()=> {


        const total={total_pages:2}  
        const current={page:1}

        const wrapper = mount (<PaginatorComponent  total={total} current={current} />, {context: {store}});
        
        expect(wrapper).toMatchSnapshot();
       
    });
})