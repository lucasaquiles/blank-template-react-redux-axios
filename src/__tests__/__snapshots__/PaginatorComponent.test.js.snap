// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Paginator should render  1`] = `
<Connect(PaginatorComponent)
  current={
    Object {
      "page": 1,
    }
  }
  total={
    Object {
      "total_pages": 2,
    }
  }
>
  <PaginatorComponent
    current={
      Object {
        "page": 1,
      }
    }
    paginationAction={[Function]}
    total={
      Object {
        "total_pages": 2,
      }
    }
  >
    <ul
      className="paginador"
    />
  </PaginatorComponent>
</Connect(PaginatorComponent)>
`;
